src/server/scripts/Northrend/Ulduar/Ulduar/boss_ignis.cpp

```diff
@@ -125,15 +125,18 @@ class npc_ulduar_iron_construct : public CreatureScript
             }
             else if (spell->Id == SPELL_HEAT_BUFF)
             {
-                if (Aura* a = me->GetAura(SPELL_HEAT_BUFF))
-                    if( a->GetStackAmount() >= RAID_MODE(10,20) )
+                if (Aura* heat = me->GetAura(SPELL_HEAT_BUFF))
+                {
+                    if (heat->GetStackAmount() >= 10)
+                    {
+                        if (heat->GetStackAmount() > 10)
```

Comment:
Couldn't you just do > 10 instead of having >= and > check?

Reply:
Nop, already tested because I thought the same. Doing that change, they will gain SPELL_MOLTEN at the first stack, which is horribly wrong.
In the current way of this PR, it will "consume" the 11 stack and cast molten instead of adding a stack, so this way for doing the check of the 10 stack is needed and.. fully functional as the test above confirms 🐱  !